build.gradle

```diff
@@ -724,6 +836,58 @@ subprojects {
                                 }
                             }
                             projectMakePatches.dependsOn "project${child.name}MakePatches"
+
+                            //Adapted from the mcpmap task
+                            task "mojmap${child.name}Decompile"(dependsOn: [generateMojMap, "project${child.name}ApplyPatches"]) {
+                                def out_path = PATH_MOJMAP + child.name.toLowerCase() + '/'
+                                def input_path = tasks.getByName("project${child.name}ApplyPatches").target.toPath()
+
+                                inputs.file file(PATH_MOJMAP + "mappings.zip")
+                                doFirst {
+                                    def names = [:]
+                                    new ZipFile(file(PATH_MOJMAP + "mappings.zip")).withCloseable { zip ->
+                                        zip.entries().findAll { !it.directory && it.name.endsWith('.csv') }
+                                                .each {
+                                                    def reader = new CsvReader()
+                                                    reader.containsHeader = true
+                                                    def csv = reader.read(new InputStreamReader(zip.getInputStream(it)))
+                                                    csv.rows.each { names[it.getField('searge') == null ? it.getField('param') : it.getField('searge')] = it.getField('name') }
+                                                }
+                                    }
+                                    Files.walk(input_path).withCloseable { stream ->
+                                        stream.each { entry ->
+                                            if(!entry.toFile().isDirectory()) {
+                                                def out = new File(out_path + entry.toString().substring(entry.toString().indexOf("src\\main\\java\\")))
```

Comment:
This does not work off of Windows - please use the appropriate platform-native separator.